streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/python/MsgPackSerializer.java

```diff
@@ -0,0 +1,119 @@
+import java.util.List;
+import java.util.Map;
+import org.msgpack.core.MessageBufferPacker;
+import org.msgpack.core.MessagePack;
+import org.msgpack.core.MessageUnpacker;
+import org.msgpack.value.ArrayValue;
+import org.msgpack.value.FloatValue;
+import org.msgpack.value.IntegerValue;
+import org.msgpack.value.MapValue;
+import org.msgpack.value.Value;
+
+public class MsgPackSerializer {
+
+  public byte[] serialize(Object obj) {
+    MessageBufferPacker packer = MessagePack.newDefaultBufferPacker();
+    serialize(obj, packer);
+    return packer.toByteArray();
+  }
+
+  private void serialize(Object obj, MessageBufferPacker packer) {
+    try {
+      if (obj == null) {
+        packer.packNil();
+      } else {
+        Class<?> clz = obj.getClass();
+        if (clz == Boolean.class) {
+          packer.packBoolean((Boolean) obj);
+        } else if (clz == Integer.class) {
+          packer.packInt((Integer) obj);
+        } else if (clz == Long.class) {
+          packer.packLong((Long) obj);
+        } else if (clz == Double.class) {
+          packer.packDouble((Double) obj);
+        } else if (clz == byte[].class) {
+          byte[] bytes = (byte[]) obj;
+          packer.packBinaryHeader(bytes.length);
+          packer.writePayload(bytes);
+        } else if (clz == String.class) {
+          packer.packString((String) obj);
+        } else if (obj instanceof Collection) {
+          Collection collection = (Collection) (obj);
+          packer.packArrayHeader(collection.size());
+          for (Object o : collection) {
+            serialize(o, packer);
+          }
+        } else if (obj instanceof Map) {
+          Map map = (Map) (obj);
+          packer.packMapHeader(map.size());
+          for (Object o : map.entrySet()) {
+            Map.Entry e = (Map.Entry) o;
+            serialize(e.getKey(), packer);
+            serialize(e.getValue(), packer);
+          }
+        } else {
+          throw new UnsupportedOperationException("Unsupported type " + clz);
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Serialize error for object " + obj, e);
+    }
+  }
+
+  public Object deserialize(byte[] bytes) {
+    try {
+      MessageUnpacker unpacker = MessagePack.newDefaultUnpacker(bytes);
+      return convert(unpacker.unpackValue());
+    } catch (Exception e) {
+      String hex = BaseEncoding.base16().lowerCase().encode(bytes);
+      throw new RuntimeException("Deserialize error: " + hex, e);
+    }
+  }
+
+  private Object convert(Value value) {
+    switch (value.getValueType()) {
+      case NIL:
+        return null;
+      case BOOLEAN:
+        return value.asBooleanValue().getBoolean();
+      case INTEGER:
+        IntegerValue iv = value.asIntegerValue();
+        if (iv.isInIntRange()) {
+          return iv.toInt();
+        } else if (iv.isInLongRange()) {
+          return iv.toLong();
+        } else {
+          return iv.toBigInteger();
+        }
+      case FLOAT:
```

Comment:
`org.ray.streaming.runtime.python.MsgPackSerializer#serialize(java.lang.Object, org.msgpack.core.MessageBufferPacker)` doesn't allow support serialize `float`, so that we don't have this issue.

Reply:
```
 else if (clz == Double.class) {
          packer.packDouble((Double) obj);
        }
```
msgpack doesn't support single-presion float